--- CA.pl.in 2016-08-26 18:53:10.488416865 +0200
+++ CA.pl.in 2016-08-26 18:53:44.615351929 +0200
@@ -33,7 +33,7 @@
 my $PKCS12 = "$openssl pkcs12";
 
 # default openssl.cnf file has setup as per the following
-my $CATOP = "./demoCA";
+my $CATOP = "/etc/ssl";
 my $CAKEY = "cakey.pem";
 my $CAREQ = "careq.pem";
 my $CACERT = "cacert.pem";
--- openssl.cnf 2016-08-26 18:51:08.737404963 +0200
+++ openssl.cnf 2016-08-26 18:52:12.027942905 +0200
@@ -39,7 +39,7 @@
 ####################################################################
 [ CA_default ]
 
-dir        = ./demoCA      # Where everything is kept
+dir        = /etc/ssl      # Where everything is kept
 certs      = $dir/certs        # Where the issued certs are kept
 crl_dir        = $dir/crl      # Where the issued crl are kept
 database   = $dir/index.txt    # database index file.
